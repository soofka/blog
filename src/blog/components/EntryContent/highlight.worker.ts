export default () => {
  self.addEventListener('message', (event) => {
    try {
      importScripts('https://soofka.pl/scripts/highlight/highlight.pack.js');
      const result = self.hljs.highlightAuto(event.data[0]);
      postMessage({ code: result.value, codeBlockIndex: event.data[1] });
    } catch (e) {
      postMessage({ ...e });
    }
  });
};
