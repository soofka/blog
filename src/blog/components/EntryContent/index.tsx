import * as React from 'react';

// import highlightWorker from './highlight.worker';
// import WebWorker from 'common/WebWorker';

interface EntryContentPropsInterface {
  content: string;
}

declare const hljs: any;

export class EntryContent extends React.Component<EntryContentPropsInterface> {

  componentDidMount(): void {
    this.highlightCodeBlocks();
  }

  highlightCodeBlocks(): void {
    document.querySelectorAll('pre code').forEach((codeBlock: Node) => {
      hljs.highlightBlock(codeBlock);
    });
  }

  render(): JSX.Element {
    const { content } = this.props;

    return (
      <div dangerouslySetInnerHTML={{ __html: content }}></div>
    );
  }

}

export default EntryContent;
