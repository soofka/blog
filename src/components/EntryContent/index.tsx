import * as React from 'react';

interface EntryContentPropsInterface {
  content: string;
}

export const EntryContent = (props: EntryContentPropsInterface): JSX.Element => {
  const { content } = props;

  return (
    <div dangerouslySetInnerHTML={{ __html: content }}></div>
  );
};

export default EntryContent;
