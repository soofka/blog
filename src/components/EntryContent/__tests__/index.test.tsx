import * as React from 'react';
import { shallow } from 'enzyme';

import EntryContent from 'components/EntryContent';

describe('<EntryContent/>', () => {

  it('renders correctly', () => {
    expect(shallow(<EntryContent content="testContent" />)).toMatchSnapshot();
  });

});
