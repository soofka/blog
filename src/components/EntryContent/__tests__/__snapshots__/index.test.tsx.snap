// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryContent/> renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "testContent",
    }
  }
/>
`;
